torch/csrc/distributed/autograd/context/dist_autograd_context.h

```diff
@@ -40,9 +42,23 @@ class TORCH_API DistAutogradContext {
   DistAutogradContext(DistAutogradContext&&) = delete;
   DistAutogradContext& operator=(DistAutogradContext&&) = delete;
 
+  static_assert(
```

Comment:
@pietern , not sure if this is what you meant by adding the `static_assert`. This asserts that the `worker_id_t` range is at most the range of `int16_t`, so for example if `worker_id_t` changed to `int32`, this assert would fail. In terms of memory constraints, not sure what limit we want to set here (I just went with the current limit of `worker_id_t` for now, but maybe we want `int32` for example).

Reply:
Yes, this is exactly what I meant. The purpose of `static_assert` here is to let the developer know that existing assumptions have changed, and that that may cause issues. At least if this happens, there won't be any unexpected surprises.